repl/src/main/scala/ammonite/repl/interp/Preprocessor.scala

```diff
@@ -99,30 +99,36 @@ object Preprocessor{
 
     val Expr = Processor{
       //Expressions are lifted to anon function applications so they will be JITed
-      case (name, code, tree) => Output(s"val $name = \n$code", Seq(pprint(name)))
+      case (name, code, tree) => Output(s"val $name = $code", Seq(pprint(name)))
     }
 
     val decls = Seq[(String, String, G#Tree) => Option[Preprocessor.Output]](
       ObjectDef, ClassDef, TraitDef, DefDef, TypeDef, PatVarDef, Import, Expr
     )
 
-    def apply(stmts: Seq[String], wrapperId: String): Res[Preprocessor.Output] = {
+    def apply(stmts: Seq[String],
+              wrapperId: String,
+              comment: String = ""): Res[Preprocessor.Output] = {
       val unwrapped = stmts.flatMap{x => Parsers.unwrapBlock(x) match {
-        case Some(contents) => Parsers.split(contents).get.get.value
+        case Some(contents) =>
+          Parsers.split(contents).get.get.value
+
         case None => Seq(x)
       }}
       unwrapped match{
         case Nil => Res.Skip
-        case postSplit => complete(stmts.mkString, wrapperId, postSplit.map(_.trim))
+        case postSplit =>
+          complete(stmts.mkString(""), wrapperId, postSplit, comment)
+
       }
     }
     
-    def complete(code: String, wrapperId: String, postSplit: Seq[String]) = {
+    def complete(code: String, wrapperId: String, postSplit: Seq[String], comment: String) = {
       val reParsed = postSplit.map(p => (parse(p), p))
       val errors = reParsed.collect{case (Left(e), _) => e }
       if (errors.length != 0) Res.Failure(None, errors.mkString("\n"))
       else {
-        val allDecls = for (((Right(trees), code), i) <- reParsed.zipWithIndex) yield {
+        val allDecls = for (((Right(trees @ x :: y), code), i) <- reParsed.zipWithIndex) yield {
```

Comment:
Otherwise it can take empty trees also and was giving error accessing trees(0) in this line
`   case trees if trees.forall(_.isInstanceOf[G#Import]) => handleTree(trees(0)) `